Adding xan from md

fredrik-hammar opened the pull request:
I gave a shot at implementing my suggestion for support of Markdown table in the `from` command (#621).

- This brings in the `markdown` crate as a dependency.
- I avoided release work like generating doc, changelog, or version dump

## Testing

Tested with attached [test.md](https://github.com/user-attachments/files/20164172/test.md).
```sh
cargo run from test.md|xan view
```

Also, with empty input.
```sh
cargo run from -f md <<<""
```

Yomguithereal commented on src/cmd/from.rs:
```diff
@@ -36,6 +37,7 @@ impl SupportedFormat {
             "txt" | "text" | "lines" => Self::Text,
             "npy" => Self::Npy,
             "tar" | "tar.gz" => Self::Tar,
+            "md" => Self::Md,
```
I think we should also allow `markdown` here for better UX

fredrik-hammar replied:
Sure, I needed to realign the usage message table for this.

Yomguithereal commented on Cargo.toml:
```diff
@@ -113,6 +113,7 @@ uuid = { version = "1.16.0", features = ["v4"] }
 # NOTE: this is necessary until the `zip` crate fix their semver violation
 # or until both `calamine` and `rust_xlsxwriter` pin their dependency.
 zip = "=2.5.0"
+markdown = "1.0.0"
```
Move this up in the above list in alphabetical order :)

fredrik-hammar replied:
Done!

Yomguithereal commented:
@fredrik-hammar looks good to me. Do you think it could be useful to add something like a `xan from md` specific flag selecting the nth table in given document?

Yomguithereal commented:
Also, since we are pulling the markdown crate, can I ask you to check the code related to xan to md to tell me whether we could use it to make the code more robust there?

fredrik-hammar commented:
> @fredrik-hammar looks good to me. Do you think it could be useful to add something like a `xan from md` specific flag selecting the nth table in given document?

I was thinking about this too, but wasn't sure if it's would be the best way to specify a location in Markdown. I now decided to go with a `--select` option, modelled after the select command. This way, you can easily choose first or last table easily with `-1`, and allows for future extensions for other types of selectors.

The code got a little messy dealing with underflows and nice error messages similar to select if index is out of bounds. Please let me know if it should be split up somehow.

fredrik-hammar commented:
> Also, since we are pulling the markdown crate, can I ask you to check the code related to xan to md to tell me whether we could use it to make the code more robust there, especially wrt escaping?

Sure, I'll give it a look. I was going to suggest some option to disable escaping as well.

fredrik-hammar commented:
> I think it would be better to use another name for this flag like `--nth-table` or something like this. `-s/--select` is used throughout the tool and is always used to select columns from a CSV file and I feel this breaks user expectations here, no?

That's a fair point. I first considered `--nth`, and `--nth-table` is even better, but one problem is I felt the short option could be confused with `-n, --no-headers`, but maybe there's no risk of that for `xan to`.

> Why would you want to disable escaping? What's the use-case here?

If you have Markdown table that has a notes or comments column for instance with formatting and want to use xan to process it, and then copy back the changes, e.g. `xan from -f md ... | xan ... | xan to -f md`, then the formatting will get mangled.

fredrik-hammar commented:
Regarding using the `markdown` crate for `xan to`, it seems that `markdown` is pretty limited when it comes to creating and manipulating the AST and doesn't support formatting to Markdown, so I'm considering switching to [`comrak`](https://github.com/kivikakk/comrak) which does. 

It also seems to be more widely used, but didn't show up in my initial search for crates.

I hope you don't mind me switching to it. I'll update this PR.

Yomguithereal commented:
Which of both crates has the least dependencies and is the lighter?

fredrik-hammar commented:
> Which of both crates has the least dependencies and is the lighter?

`markdown` has the least dependencies, but `comrak` is the overall lighter one according to `cargo bloat`. I'm guessing this is because `markdown` is `no_std`.

I got it working with `comrak`, but will polish it a bit more before pushing.

Here are the outputs I checked to compare the two: 

```
$ cargo tree -p markdown
markdown v1.0.0
└── unicode-id v0.3.5
```

```
$ cargo tree -p comrak
comrak v0.39.0
├── caseless v0.2.2
│   └── unicode-normalization v0.1.24
│       └── tinyvec v1.9.0
│           └── tinyvec_macros v0.1.1
├── entities v1.0.1
├── memchr v2.7.4
├── slug v0.1.6
│   └── deunicode v1.6.2
├── typed-arena v2.0.2
└── unicode_categories v0.1.1
```

```
$ cargo bloat --filter markdown --release
File .text     Size    Crate Name
0.1%  0.1%  13.3KiB markdown markdown::to_mdast::exit
0.0%  0.1%   9.5KiB markdown markdown::to_mdast::compile
0.0%  0.0%   4.0KiB markdown markdown::construct::attention::resolve
0.0%  0.0%   3.6KiB markdown markdown::construct::gfm_autolink_literal::resolve
0.0%  0.0%   3.1KiB markdown markdown::construct::list_item::resolve
0.0%  0.0%   2.9KiB markdown markdown::construct::label_end::resolve
0.0%  0.0%   2.9KiB markdown core::ptr::drop_in_place<markdown::mdast::Node>
0.0%  0.0%   2.7KiB markdown core::ptr::drop_in_place<markdown::mdast::Node>
0.0%  0.0%   2.7KiB markdown markdown::construct::partial_mdx_expression::before
0.0%  0.0%   2.7KiB markdown markdown::parser::parse
0.0%  0.0%   2.6KiB markdown markdown::subtokenize::subtokenize
0.0%  0.0%   2.6KiB markdown markdown::construct::document::flow_end
0.0%  0.0%   2.5KiB markdown markdown::to_mdast::on_mismatch_error
0.0%  0.0%   2.5KiB markdown markdown::util::edit_map::EditMap::consume
0.0%  0.0%   2.2KiB markdown markdown::construct::heading_setext::resolve
0.0%  0.0%   2.1KiB markdown markdown::to_mdast::on_exit_label_text
0.0%  0.0%   2.0KiB markdown markdown::construct::html_flow::tag_name
0.0%  0.0%   2.0KiB markdown markdown::construct::gfm_table::resolve
0.0%  0.0%   1.8KiB markdown <markdown::mdast::Node as alloc::string::ToString>::to_string
0.0%  0.0%   1.7KiB markdown markdown::construct::partial_whitespace::resolve_whitespace
0.7%  1.5% 140.5KiB          And 517 smaller methods. Use -n N to show more.
1.0%  2.3% 209.8KiB          filtered data size, the file size is 20.9MiB
```

```
$ cargo bloat --filter comrak --release
File .text     Size  Crate Name
0.1%  0.2%  17.2KiB comrak comrak::parser::inlines::Subject::parse_inline
0.1%  0.2%  14.5KiB comrak comrak::parser::Parser::process_line
0.0%  0.1%   8.8KiB comrak comrak::parser::parse_document
0.0%  0.1%   8.5KiB comrak comrak::scanners::autolink_email
0.0%  0.1%   8.4KiB comrak comrak::scanners::html_block_start
0.0%  0.1%   5.4KiB comrak comrak::scanners::html_block_end_1
0.0%  0.1%   4.7KiB comrak comrak::parser::inlines::Subject::process_emphasis
0.0%  0.0%   4.2KiB comrak comrak::scanners::autolink_uri
0.0%  0.0%   4.1KiB comrak comrak::parser::table::try_opening_block
0.0%  0.0%   3.8KiB comrak comrak::scanners::link_title
0.0%  0.0%   3.8KiB comrak comrak::parser::Parser::resolve_reference_link_definitions
0.0%  0.0%   3.2KiB comrak comrak::scanners::html_tag
0.0%  0.0%   3.2KiB comrak comrak::scanners::html_block_start_7
0.0%  0.0%   2.9KiB comrak comrak::scanners::table_cell
0.0%  0.0%   2.8KiB comrak comrak::parser::inlines::Subject::handle_wikilink
0.0%  0.0%   2.8KiB comrak comrak::parser::autolink::process_email_autolinks
0.0%  0.0%   2.3KiB comrak comrak::scanners::open_code_fence
0.0%  0.0%   2.1KiB comrak core::slice::sort::unstable::quicksort::quicksort
0.0%  0.0%   2.0KiB comrak comrak::scanners::html_block_end_5
0.0%  0.0%   2.0KiB comrak comrak::scanners::html_block_end_2
0.3%  0.7%  68.4KiB        And 147 smaller methods. Use -n N to show more.
0.8%  1.9% 175.0KiB        filtered data size, the file size is 20.4MiB
```

fredrik-hammar commented:
I've switched to `comrak` and renamed option to `--nth-table` so I'm done with this PR unless there is anything else.

I looked into using `comrak` to format table in `xan to md`, but it currently does not support pretty printing tables, see <https://github.com/kivikakk/comrak/discussions/434>.  I'm considering taking a stab at adding that support and getting back to this later.

Yomguithereal commented:
@fredrik-hammar will merge now :)
